src/kakarot/accounts/library.cairo

```diff
@@ -15,64 +18,311 @@ from starkware.starknet.common.syscalls import (
     storage_read,
     storage_write,
     StorageReadRequest,
+    CallContract,
+    get_tx_info,
+    get_contract_address,
+    get_caller_address,
+    replace_class,
 )
 from starkware.cairo.common.memset import memset
+from utils.eth_transaction import EthTransaction
 
 from kakarot.interfaces.interfaces import IERC20, IKakarot
+from kakarot.errors import Errors
 
+// @dev: should always be zero for EOAs
 @storage_var
-func bytecode_len_() -> (res: felt) {
+func Account_bytecode_len() -> (res: felt) {
 }
 
 @storage_var
-func storage_(key: Uint256) -> (value: Uint256) {
+func Account_storage(key: Uint256) -> (value: Uint256) {
 }
 
 @storage_var
-func is_initialized_() -> (res: felt) {
+func Account_is_initialized() -> (res: felt) {
 }
 
 @storage_var
-func evm_address() -> (evm_address: felt) {
+func Account_evm_address() -> (evm_address: felt) {
 }
 
 @storage_var
-func nonce() -> (nonce: felt) {
+func Account_nonce() -> (nonce: felt) {
+}
+
+@storage_var
+func Account_kakarot_address() -> (kakarot_address: felt) {
+}
+
+@storage_var
+func Account_implementation() -> (address: felt) {
+}
+
+@event
+func transaction_executed(response_len: felt, response: felt*, success: felt, gas_used: felt) {
 }
 
-// Define the number of bytes per felt
 const BYTES_PER_FELT = 31;
 
-// @title ContractAccount main library file.
-// @notice This file contains the EVM smart contract account representation logic.
-namespace ContractAccount {
+// @title Account main library file.
+// @notice This file contains the EVM account representation logic.
+// @dev: Both EOAs and Contract Accounts are represented by this contract.
+namespace GenericAccount {
```

Comment:
`Account` is already a "reserved" keyword for internal account representation. What other name do you suggest?

Reply:
mh indeed there is the "storage" type Account and the "contract" type Account. Does this create a name clash ?

Reply:
I dont think it will (both should not be imported in the same files) but I prefer not to have two similar namespaces in the codebase. perhaps `AccountContract`?